### PR TITLE
TEST/UD: Fix crep_drop2 test - validate connect after hook is restored.

### PR DESCRIPTION
**Pull request by yosefe**
Update for #1201

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/932/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2898/ for details (Mellanox internal link).

File: test/gtest/uct/test_ud.cc [hunk 1]
```diff
@@ -378,15 +378,16 @@ UCS_TEST_P(test_ud, crep_drop2) {
...
 
-    validate_connect(ep(m_e1), 0U);
-    validate_connect(ep(m_e2), 0U);
+    short_progress_loop(100);
 
     /* Remove filter for CREP to be handled and TX win to be freed. */
     ep(m_e1)->rx.rx_hook = uct_ud_ep_null_hook;
     ep(m_e2)->rx.rx_hook = uct_ud_ep_null_hook;
 
+    validate_connect(ep(m_e1), 0U);
```
Comment (brminich) [inline test/gtest/uct/test_ud.cc]: Now validate_connect (which invokes progress) is called when CREP drop filter is removed. It seems that the situation  when both eps sent CREQ and CREP is possible. Then the following check for acked_psn == 1 may fail.

Comment (yosefe) [inline test/gtest/uct/test_ud.cc]: @brminich you are right, and it also reproduced after ~500 iterations. i will add a check for CREQ_NOTSENT flag.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/936/ for details.

File: test/gtest/uct/test_ud.cc [hunk 2]
```diff
@@ -389,10 +399,14 @@ UCS_TEST_P(test_ud, crep_drop2) {
     validate_connect(ep(m_e2), 0U);
 
     /* Expect that creq (and maybe crep already) are sent */
-    EXPECT_EQ(ep(m_e1)->tx.acked_psn, 1);
-    EXPECT_EQ(ep(m_e2)->tx.acked_psn, 1);
-    EXPECT_GE(ep(m_e1)->tx.psn,       2);
-    EXPECT_GE(ep(m_e2)->tx.psn,       2);
+    if (!(ep(m_e1)->flags & UCT_UD_EP_FLAG_CREQ_NOTSENT)) {
+        EXPECT_EQ(ep(m_e1)->tx.acked_psn, 1);
```
Comment (brminich) [inline test/gtest/uct/test_ud.cc]: maybe use  new validate_send here?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2902/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2903/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/937/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/938/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/939/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/940/ for details.

**Comment by yosefe** [comment]
@brminich pls take a look

**Comment by brminich** [comment]
What if one of the eps will send CREP after receiving CREP from its peer. It seems that acked_psn can be 2 in this case.

**Comment by yosefe** [comment]
@brminich will change EXPECT_EQ in validate_send() to EXPECT_GE

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/942/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2907/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich pls take a look

**Comment by brminich** [comment]
:+1:

**Pull request merged**